### PR TITLE
Completely remove need to vendor pkg/build/builder

### PR DESCRIPTION
**Pull request by smarterclayton**
In preparation for moving pkg/build/builder out of the repository, collapse all prerequisites down by eliminating version skews.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@deads2k @bparees unwinding this now will help during 3.10 (it's on my list to pull it out as soon as we have image builds from aos-cd-jobs).

**Comment by smarterclayton** [comment]
Note that https://github.com/openshift/origin/pull/18740/commits/79a6ace1b2d4d5e928214d0d133f8a666ec12abf was an UPSTREAM carry that got accidentally glommed into our code.

**Comment by deads2k** [comment]
This is good stuff for builders to do, but this actually makes our next rebase more risky, not less.  I'd rather keep them in their own world and split them off than end up with the "whose version is more important".

@liggitt it's your rebase, but I wouldn't want this risk in mine.

**Comment by smarterclayton** [comment]
Kube 1.9 and 1.10 are at docker/docker@4f3616fb1c112e206b88cb7a9922bf49067a7756 (july 2017).  This bumps docker/docker@b68221c37ee597950364788204546f9c9d0e46a1 which is august 2017.  All 1.9 tests pass in origin.  Going to test this against kube 1.10 as a PR for giggles.

File: pkg/oc/bootstrap/docker/up.go [hunk 1]
```diff
@@ -722,6 +724,10 @@ func getDockerClient(out io.Writer, dockerMachine string, canStartDockerMachine
 	if err != nil {
 		return nil, errors.ErrNoDockerClient(err)
 	}
+	// negotiate the correct API version with the server
+	ctx, fn := context.WithTimeout(context.Background(), 100*time.Millisecond)
```
Comment (liggitt) [inline pkg/oc/bootstrap/docker/up.go]: bump this to a few seconds, maybe

File: pkg/build/builder/sti_test.go [hunk 2]
```diff
@@ -57,7 +58,7 @@ func newTestS2IBuilder(config testS2IBuilderConfig) *S2IBuilder {
 		&FakeDocker{
 			errPushImage: config.errPushImage,
 		},
-		"unix:///var/run/docker.sock",
+		"unix:///var/run/docker2.sock",
```
Comment (liggitt) [inline pkg/build/builder/sti_test.go]: ?

File: pkg/build/builder/sti_test.go [hunk 3]
```diff
@@ -72,6 +73,10 @@ func newTestS2IBuilder(config testS2IBuilderConfig) *S2IBuilder {
 func makeBuild() *buildapiv1.Build {
 	t := true
 	return &buildapiv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
```
Comment (liggitt) [inline pkg/build/builder/sti_test.go]: ?

File: pkg/build/builder/common.go [hunk 4]
```diff
@@ -93,7 +93,7 @@ func buildInfo(build *buildapiv1.Build, sourceInfo *git.SourceInfo) []KeyValue {
 // that the built image can be referred to unambiguously even in the face of
 // concurrent builds with the same name in the same namespace.
 func randomBuildTag(namespace, name string) string {
-	repo := fmt.Sprintf("docker.io/%s/%s", namespace, name)
+	repo := fmt.Sprintf("b.openshift.io/%s/%s", namespace, name)
```
Comment (liggitt) [inline pkg/build/builder/common.go]: can we make this more obviously temporary, like `tmp.build.openshift.io/...`?

File: glide.yaml [hunk 5]
```diff
@@ -162,6 +161,15 @@ import:
 # fsouza and to limit dependencies
 - package: github.com/docker/docker
   version: b68221c37ee597950364788204546f9c9d0e46a1
+# keep us up to date with latest DNS security fixes
+- package: github.com/miekg/dns
+  version: master
+# etcd pins a very old version that has contention issues
+- package: github.com/google/btree
+  version: master
```
Comment (liggitt) [inline glide.yaml]: why not a specific commit?

File: pkg/build/builder/sti_test.go (hunk 2 above)
Comment (bparees) [reply pkg/build/builder/sti_test.go]: +1

File: pkg/build/builder/sti_test.go (hunk 3 above)
Comment (bparees) [reply pkg/build/builder/sti_test.go]: also +1

File: pkg/build/builder/common.go (hunk 4 above)
Comment (bparees) [reply pkg/build/builder/common.go]: @liggitt gets all the +1s today.

**Comment by bparees** [comment]
just the things @liggitt already noted.  i leave dependency management to the experts.

File: glide.yaml (hunk 5 above)
Comment (smarterclayton) [reply glide.yaml]: they are good at their PI management

File: pkg/build/builder/common.go (hunk 4 above)
Comment (smarterclayton) [reply pkg/build/builder/common.go]: We'll hit length limits at some point.  That's probably fine.

File: pkg/build/builder/sti_test.go (hunk 2 above)
Comment (smarterclayton) [reply pkg/build/builder/sti_test.go]: To ensure this doesn't go out to the docker socket if someone accidentally exposes a path

File: pkg/build/builder/sti_test.go (hunk 3 above)
Comment (smarterclayton) [reply pkg/build/builder/sti_test.go]: Builds without these fields are not valid and cause tests to report incorrect values

**Comment by smarterclayton** [comment]
Confirmed the bump didn't break kube 1.10 in kubernetes/kubernetes#60453.

Other comments addressed, including breaking the cri-o dependency.

**Comment by smarterclayton** [comment]
Unblocking the image PR now.

**Pull request merged**